mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -487,6 +487,27 @@ void ModuleImport::addDebugIntrinsic(llvm::CallInst *intrinsic) {
   debugIntrinsics.insert(intrinsic);
 }
 
+LogicalResult ModuleImport::convertLinkerOptionsMetadata() {
+  for (const llvm::NamedMDNode &named : llvmModule->named_metadata()) {
+    if (named.getName() != "llvm.linker.options")
+      continue;
+    // llvm.linker.options operands are lists of strings.
+    for (const llvm::MDNode *md : named.operands()) {
+      SmallVector<StringRef> options;
+      for (const llvm::MDOperand &option : md->operands()) {
+        if (auto str = dyn_cast_or_null<llvm::MDString>(option))
```

Comment:
Can it ever be null?

Reply:
Not if the LLVM IR is valid; I wasn’t sure if it was verified another way before going through the import or if I should do it here. I can change this to just a cast if  that’s the case.

Reply:
You can assume verified LLVM IR in the import.